snapshot/db.go

```diff
@@ -0,0 +1,45 @@
+package snapshot
+
+// ID identifies a value in DB
+type ID string
+
+// DB is the Scoot Database. It holds Values identified by an ID.
+// A Value may be a Snapshot or a Revision.
+type DB interface {
+	// Ingest
+
+	// IngestDir ingests a directory directly.
+	// The created value is a Snapshot.
+	IngestDir(dir string) (ID, error)
+
+	// IngestGitCommit ingests the commit identified by commitish from ingestRepo
+	// commitish may be any string that identifies a commit
+	// The created value is a Revision.
+	IngestGitCommit(ingestRepo *repo.Repository, commitish string) (SnapID, error)
+
+	// Operations
+
+	// SnapshotForRevision unwraps a Revision and returns a Snapshot.
+	// Errors if id does not identify a Revision.
+	SnapshotForRevision(id ID) (ID, error)
```

Comment:
I added a more holistic description of the structure in doc.go. (Also, renamed Revision to SnapshotWithHistory).

Unwraps means to create a Snapshot without History.

You mean saying it returns a Snapshot ID as opposed to saying "Snapshot"?

There's only one ID input parameter.